File: japiwznm/WznmQRlsList.java
/**
	* \file WznmQRlsList.h
	* Java API code for record of table TblWznmQRlsList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQRlsList {
	
	public WznmQRlsList(
				int jnum
				, String sref
				, String stubRefWznmMComponent
				, String stubRefWznmMMachine
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubRefWznmMComponent = stubRefWznmMComponent;
		this.stubRefWznmMMachine = stubRefWznmMMachine;
	};

	public int jnum;
	public String sref;
	public String stubRefWznmMComponent;
	public String stubRefWznmMMachine;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQRlsList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefWznmMComponent = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMComponent", "cmp", null, 0);
			stubRefWznmMMachine = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMMachine", "mch", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQRlsList.cpp
/**
	* \file WznmQRlsList.cpp
	* API code for table TblWznmQRlsList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQRlsList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQRlsList
 ******************************************************************************/

WznmQRlsList::WznmQRlsList(
			const uint jnum
			, const string sref
			, const string stubRefWznmMComponent
			, const string stubRefWznmMMachine
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->stubRefWznmMComponent = stubRefWznmMComponent;
	this->stubRefWznmMMachine = stubRefWznmMMachine;
};

bool WznmQRlsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQRlsList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefWznmMComponent", "cmp", stubRefWznmMComponent);
		extractStringUclc(docctx, basexpath, "stubRefWznmMMachine", "mch", stubRefWznmMMachine);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQRlsList
 ******************************************************************************/

ListWznmQRlsList::ListWznmQRlsList() {
};

ListWznmQRlsList::ListWznmQRlsList(
			const ListWznmQRlsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRlsList(*(src.nodes[i]));
};

ListWznmQRlsList& ListWznmQRlsList::operator=(
			const ListWznmQRlsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRlsList(*(src.nodes[i]));

	return *this;
};

ListWznmQRlsList::~ListWznmQRlsList() {
	clear();
};

void ListWznmQRlsList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQRlsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQRlsList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQRlsList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQRlsList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQRlsList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQRlsList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
